copy/copy.go

```diff
@@ -1076,6 +1076,28 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 			}
 			return blobInfo, cachedDiffID, nil
 		}
+
+		rDest, ok := ic.c.dest.(types.ImageDestinationReusableWithRef)
+		if ok {
+			reused, blobInfo, err = rDest.TryReusingBlobWithRef(ctx, srcInfo, srcRef)
```

Comment:
We should call only one of the `TryReusingBlob` methods, so that a transport can implement both (to be usable by old clients as well as new) and we don’t try to reuse everything twice. That would also reduce duplication in the consumption of the results.

Reply:
Fixed it by making `TryReusingBlobWithSrcRef` be a superset of `TryReusingBlob`.